src/lib.rs

```diff
@@ -99,22 +103,57 @@ impl fmt::Display for Repository {
 
 pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     let mut builder = ClientBuilder::new();
+    let mut print_auth_warning = false; 
 
     if let Some(ref token) = config.token {
         builder.set_authorization_token(token.to_owned());
     }
+    else {
+        print_auth_warning = true; 
+    }
 
     let client = builder.build()?;
 
     let mut stars: Vec<Star> = Vec::new();
 
     let mut next_link = config.url();
+    
+    let mut remaining: i32 = 0; 
+    let mut total: i32 = 0;
+    let mut reset_time = SystemTime::now();
 
     while next_link.is_some() {
         if let Some(link) = next_link {
             let mut res = client.get(&link).send()?;
-            next_link = extract_link_next(res.headers());
 
+            for header in res.headers().iter() {
+                match header.name() {
+                    "X-RateLimit-Limit" => {
+                        total = header.value_string().parse::<i32>()?;
+                    },
+                    "X-RateLimit-Remaining" => { 
+                        remaining = header.value_string().parse::<i32>()?;
+                    },
+                    "X-RateLimit-Reset" => {
+                        let seconds = header.value_string().parse::<u64>()?;
+
+                        // Creates a new SystemTime from the specified number of whole seconds
+                        reset_time = UNIX_EPOCH + Duration::from_secs(seconds);
+                    },
+                    _ =>(),
+                }
+            }
+            
+            match res.status() {
+                StatusCode::Forbidden => {
+                    //this type of err, or panic? 
+                    return Err(format!("Uh-oh! You have {} out of {} requests remaining. Your request limit will reset in {} minutes.", remaining, total, reset_time.duration_since(SystemTime::now())?.as_secs()/60).into());
```

Comment:
Not thrilled with the [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29) `60` here (and elsewhere).

If we switched over to using `chrono` types (see the `use chrono` line, we're already using this library elsewhere) for all of the datetime work, we could use the [strftime format syntax](https://docs.rs/chrono/0.4.6/chrono/format/strftime/index.html) which I think would be a lot clearer.